integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkGenericIntegrationTest.java

```diff
@@ -115,6 +116,43 @@ void sparkEmitsApplicationLevelEvents() {
         .hasSize(1);
   }
 
+  @Test
+  @SneakyThrows
+  void sparkGathersMetrics() {
+    Dataset<Row> df = createTempDataset();
+
+    Dataset<Row> agg = df.groupBy("a").count();
+    agg.write().mode("overwrite").csv("/tmp/test_data/test_output/");
+    Thread.sleep(1000);
```

Comment:
It's preferred to use await library to avoid `Thread.sleep`. Is there a reason for doing it this way?

Reply:
Removed

Reply:
Removed